src/main/java/co/rsk/federate/signing/hsm/config/PowHSMConfig.java

```diff
@@ -0,0 +1,154 @@
+package co.rsk.federate.signing.hsm.config;
+
+import static co.rsk.federate.signing.hsm.config.PowHSMConfigParameter.*;
+
+import co.rsk.federate.signing.hsm.SignerException;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.federate.signing.config.SignerConfig;
+import co.rsk.federate.signing.hsm.HSMClientException;
+import co.rsk.federate.signing.hsm.HSMUnsupportedTypeException;
+import co.rsk.federate.signing.hsm.client.HSMBookkeepingClient;
+import com.typesafe.config.Config;
+import java.math.BigInteger;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PowHSMConfig {
+
+  private static final Logger logger = LoggerFactory.getLogger(PowHSMConfig.class);
+  private static final String SIGNER_TYPE = "hsm";
+
+  private final Config config;
+
+  private PowHSMConfig(Config config) {
+    this.config = config;
+  }
+
+  public static PowHSMConfig from(SignerConfig signerConfig) throws SignerException {
```

Comment:
Updated

Reply:
Since it is approved and updated this comment, let me merge this one and direct you to this new [PR](https://github.com/rsksmart/powpeg-node/pull/278) which is about refactoring the tests.